sql-plugin/src/main/scala/com/nvidia/spark/rapids/PrioritySemaphore.scala

```diff
@@ -27,14 +29,18 @@ class PrioritySemaphore[T](val maxPermits: Int)(implicit ordering: Ordering[T])
   private val lock = new ReentrantLock()
   private var occupiedSlots: Int = 0
 
-  private case class ThreadInfo(priority: T, condition: Condition, numPermits: Int) {
+  private case class ThreadInfo(priority: T, condition: Condition, numPermits: Int, taskId: Long) {
     var signaled: Boolean = false
   }
 
   // We expect a relatively small number of threads to be contending for this lock at any given
   // time, therefore we are not concerned with the insertion/removal time complexity.
   private val waitingQueue: PriorityQueue[ThreadInfo] =
-    new PriorityQueue[ThreadInfo](Ordering.by[ThreadInfo, T](_.priority).reverse)
+    new PriorityQueue[ThreadInfo](
+      // use task id as tie breaker when priorities are equal (both are 0 because never hold lock)
+      Ordering.by[ThreadInfo, T](_.priority).reverse.
```

Comment:
nit: could we write this as
```
Ordering.by[ThreadInfo, T](t => (t.priority, t.taskId)).reverse
```
(technically this would flip the taskId comparison but I don't think we care)

Reply:
I can see the argument for wanting to be more explicit with `thenComparing` so that's totally fine too

Reply:
I think it matters ?  We hope tasks with smaller taskid could have higher priority, so that we can avoid the very long tasks spanning from the start of stage to end of stage.